Source/WebCore/display/css/DisplayBoxDecorationPainter.cpp

```diff
@@ -1259,6 +1259,16 @@ void BoxDecorationPainter::paintBoxShadow(PaintingContext& paintingContext, Shad
     bool hasBorderRadius = m_box.hasBorderRadius();
 
     bool hasOpaqueBackground = m_box.style().backgroundColor().isOpaque();
+    
+    auto resolvedColor = [](const StyleColor& styleColor) {
+        if (styleColor.isAbsoluteColor())
+            return styleColor.absoluteColor();
+        
+        // FIXME: StyleColor should have been properly resolved already,
+        // but unfortunately it's not enforced in the type system.
+        ASSERT_NOT_REACHED();
+        return Color { };
```

Comment:
How would you fix this FIXME? Maybe give some information about why this is not happening.

Since the ASSERT_NOT_REACHED says this is not happening you could just have ASSERT(!styleColor.isAbsoluteColor()).

Reply:
I think we should just call `absoluteColor()` at every call site and `ASSERT(styleColor.isAbsoluteColor())` at the beginning of the function, instead of having the `resolveColor()` lambda.

Reply:
It's not always the same StyleColor object which is passed to the lambda.